### PR TITLE
ANW-1227: MARC export related agents 'w' and 'i' subfs

### PR DESCRIPTION
**Pull request by lorawoodford**
Updates MARC authority record exports to match the following:

For 500/510/511 subfield 'i':
- if specific_relator, value of specific relator;
-  if no specific_relator, but description, value of description;  otherwise, 
- value of relator

For 500/510/511 subfield 'w':
-   if relator = is_earlier_form_of, `a`;
-  if relator = is_later_form of, `b`;
-  if specific_relator = Acronym, `d`;
-  if specific_relator = Musical composition, `f`;
-  if specific_relator = Broader term , `g`;
-  if specific_relator = Narrower term, `h`;   otherwise, 
- `r`

Also, corrected bug wherein subfields 'w' and 'i' were being created for 100/110/111 and 400/410/411 tags. 

## Related JIRA Ticket or GitHub Issue

https://archivesspace.atlassian.net/browse/ANW-1227

## How Has This Been Tested?

Several new tests added, including a helper method to setup some enumeration values for specific relator (which is, by default, empty).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have authority to submit this code.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/app/exporters/serializers/marc_auth.rb [hunk 1]
```diff
@@ -360,10 +360,14 @@ def names_person(primary, not_primary, parallel, xml)
     end
   end
 
-  def person_name_subtags(name, xml)
+  def person_name_subtags(name, xml, related = false)
+    if related
```
Comment (mark-cooper) [inline backend/app/exporters/serializers/marc_auth.rb:364]: Not a huge deal: the `if related` section is duplicated (in person, fam, corp) so could be refactored to a method.

File: backend/spec/agent_spec_helper.rb [hunk 2]
```diff
@@ -43,4 +43,11 @@ def add_gender_values
   ge.add_enumeration_value(:value => 'female')
   ge.add_enumeration_value(:value => 'male')
   ge.add_enumeration_value(:value => 'non-binary')
-end
\ No newline at end of file
+end
+
+def add_specific_relator_values
+	spec_rel = Enumeration.find(:name => 'agent_relationship_specific_relator')
```
Comment (mark-cooper) [inline backend/spec/agent_spec_helper.rb:49]: Minor: formatting got a bit weird here. Cleanup?

File: backend/app/exporters/serializers/marc_auth.rb (hunk 1 above)
Comment (lorawoodford) [reply backend/app/exporters/serializers/marc_auth.rb]: @mark-cooper Updated.

File: backend/spec/agent_spec_helper.rb (hunk 2 above)
Comment (lorawoodford) [reply backend/spec/agent_spec_helper.rb]: @mark-cooper Got it.

**Review by mark-cooper — APPROVED** [review]

**Pull request merged**